shopfloor_reception/services/reception.py

```diff
@@ -846,6 +1014,51 @@ def _check_expiry_date(self, line):
         if use_expiration_date and not line.expiration_date:
             return self.msg_store.expiration_date_missing()
 
+    def _set_quantity__get_handlers_by_type(self):
+        return {
+            "product": self._set_quantity__by_product,
+            "packaging": self._set_quantity__by_packaging,
+            "package": self._set_quantity__by_package,
+            "location": self._set_quantity__by_location,
+            "lot": self._set_quantity__by_lot,
+        }
+
+    def _set_quantity__by_barcode(
+        self, picking, selected_line, barcode, confirmation=False
+    ):
+        handlers_by_type = self._set_quantity__get_handlers_by_type()
+        search = self._actions_for("search")
+        search_result = search.find(barcode, handlers_by_type.keys())
+        # handler = handlers_by_type.get(search_result.type, self._set_quantity__fallback)
+        handler = handlers_by_type.get(search_result.type)
+        if handler:
```

Comment:
I don't understand if you inform `search.find` which models to search for with `handlers_by_type.keys()` how could the search return another type of record ?

Reply:
it returns "none" when no record is found.
What you can see is the "fallback" behavior, when the barcode matches nothing valid for this `set_quantity` step

Reply:
ok